web/src/main/java/org/wso2/testgrid/web/api/LogEventSocketMediator.java

```diff
@@ -130,7 +132,13 @@ private void runLogTailer(TestPlan testPlan, Session session)
      * @throws TestGridException thrown when error on calculating the log file path
      */
     private Path getLogFilePath(TestPlan testPlan) throws TestGridException {
-        return TestGridUtil.getTestRunWorkspace(testPlan, false).resolve(TestGridConstants.TEST_LOG_FILE_NAME);
+        //Create logging directory
+        Path testRunDirectory = TestGridUtil.getTestRunWorkspace(testPlan);
+        String[] subStrings = testRunDirectory.toString().split(TestGridConstants.FILE_SEPARATOR, 2);
+        String logFileName = subStrings[1].replaceAll(TestGridConstants.FILE_SEPARATOR, "_")
```

Comment:
This could lead to ArrayIndexOutofBoundsException I think. We can't always guarantee that the array size is two. 

I think what you really require is the testgrid job name. I think we should pass the name from jenkinsfile into the testgrid from the job-config.yaml. wdyt?

Reply:
We will require to derive the name in all commands. Therefore, if we use the job-config.yaml, we will have to either read it in every command or persist the job name at generate-test-plan command execution. We can make use of the TestGridUtil#deriveLogFilePath for this. WDYT?